File: 724. Find Pivot Index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        sum1 = sum(nums)
        sum2 = 0
        for index, ele in enumerate(nums):
            sum1 -= ele
            if(sum1 == sum2): return index
            sum2+= ele

        return -1
        """
        #more space efficient
        sum1 = sum(nums)
        sum2 = 0
        past = 0
        index = 0
        for x in nums:
            sum2 += past
            sum1 -= x
            if(sum1 == sum2): return index
            past = x
            index +=1

        return -1
        """

File: 724. Find Pivot Index.java
class Solution {
    public int pivotIndex(int[] nums) {
        int left = 0; //fill left with the sum of the whole string
        for(int ele: nums) left += ele;

        int right = 0;
        for(int i=0; i< nums.length;i++){
            left -= nums[i];
            if(left == right) return i;
            right += nums[i];
        }
        return -1;
        
    }
}